### DISCUSSION TITLE
How to include damage in plasticity model?

### DISCUSSION THREAD

**Discussion by tstepan333** [idaholab/moose#19539] [Q&A Modules: Solid mechanics] [answered]
I'm trying to simulate a tensile test for plastic material. As a first step, I implemented a plasticity model using ComputeMultiPlasticityStress and UserObjects to define the hardening function. My goal is to capture the effect of damage on stress and stiffness. What type of material do I need to use to define a damage index?

**Comment by ngrilli** [comment, marked as answer]
Dear @tstepan333 

If you want to couple phase field damage model with plasticity model, you will need to define your own material class.
You can see an example here:
https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/FiniteStrainUObasedCPDamageVol.C
where I couple phase field damage with crystal plasticity.

Test cases are available here:
https://github.com/ngrilli/c_pfor_am/tree/main/test/tests/CPDamage
 
The idea is that you need to define the free energy and derivatives that are needed
in the phase field model for damage.

The approach is documented here:
https://aip.scitation.org/doi/abs/10.1063/1.5109761

Nicolò

**Reply by jiangwen84** [reply]
In case you want to use smeared cracking model, you can use `ComputeSmearedCrackingStress` in conjunction with any inelastic models, e.g., plasticity and creep. You can also implement your own damage models following the smeared cracking models. 

To use this, the plasticity and creep need to be defined as `StressUpdate` models which are a bit different than the UO based plasticity models. For more information, please look at https://mooseframework.inl.gov/source/materials/ComputeMultipleInelasticStress.html

**Reply by tstepan333** [reply]
@jiangwen84 Thank you for your help. Could you clarify what the UO-based plasticity model means?

**Reply by jiangwen84** [reply]
> @jiangwen84 Thank you for your help. Could you clarify what the UO-based plasticity model means?

The UserObject-based plasticity models you used. In MOOSE, we have both UO-based and stress-update based plasticity models.

**Reply by tstepan333** [reply]
Hi @jiangwen84, I need your help.
I am getting errors when using plasticity and the smeared crack model. 
<img width="954" alt="Error" src="https://user-images.githubusercontent.com/79589114/149377374-20d78a3a-624f-450b-830a-c488f0706a36.png">
In my model, the plate stretches in the x-direction, and cracks form in the y-direction. 
Do I need to specify both plasticity and smeared crack model in ComputeMultipleInelasticStress?

**Reply by loganharbour** [reply]
@tstepan333 in the future can you please post plain text not as a screenshot and instead in a code block (see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

It will then display as :
```
*** ERROR ***
The following...
```

By pasting text directly, it makes it searchable for future users experiencing the same issue.

**Reply by jiangwen84** [reply]
@tstepan333 `ComputeSmearedCrackingStress` derives from `ComputeMultipleInelasticStress`. So you only need `ComputeSmearedCrackingStress` in the input file.

**Reply by whitewalker63** [reply]
@ngrilli @jiangwen84  I’m using the ComputeMultipleinelasticstress class to compute the stress state of multiple inelastic models based on the RadialreturnStressupdate. Each of my inelastic models (which are child classes of the RadialReturnCreepStressUpdateBase) has a damage parameter/variable. How do I get the damage variable into these inelastic models? If I have to use the computeSmearedCrackingStress, do I have to first include (or call) the ScalarDamageBase in each of these inelastic modules?

**Comment by karthigesut** [comment]
@ngrilli I have a question to ask. If i want to do a damage model (crystal plasticity) on MOOSE FRAMEWORK, can i import the microstructure from NEPER? or MOOSE has the capability to create a synthetic microstructure like what NEPER and DREAM3D can do? 

Thank you. 

karthi

**Reply by ngrilli** [reply]
@karthigesut 
An example of importing Neper .msh file to run a polycrystal simulation is here:
https://github.com/ngrilli/c_pfor_am/tree/main/test/tests/neper

Hope this helps,
Nicolò

**Reply by karthigesut** [reply]
@ngrilli Thanks sir.

**Reply by karthigesut** [reply]
@ngrilli is there any free training for moose framework software? i would like to join if there is any. 

Thank you. 

Karthi

**Reply by GiudGiud** [reply]
Hello

There's the tutorial video on the website
https://mooseframework.inl.gov/index.html
and we may stream a live training in June, we're still organizing this.  it ll be advertised. 

Guillaume

**Reply by karthigesut** [reply]
@GiudGiud thanks for your reply. 

I would be happy if you can confirm the date of the training earlier. I presume its a free training. 

Thank you sir. 

Karthi
